object_store/src/lib.rs

```diff
@@ -272,7 +293,13 @@ impl AmazonS3 {
             ..Default::default()
         };
 
-        self.client.put_object(put_request).await?;
+        self.client
+            .put_object(put_request)
+            .await
+            .context(UnableToPutDataToS3 {
+                bucket: self.bucket_name.to_owned(),
```

Comment:
Does this work? (and similar changes in the rest of the error paths)

```suggestion
                bucket: self.bucket_name,
```

Reply:
Applied this change here:
https://github.com/influxdata/influxdb_iox/pull/536#discussion_r539363479
Given the explanation in the linked comment explained this not to be possible